docs/src/upgrade/release-3.8.x.asciidoc

```diff
@@ -305,6 +305,32 @@ The `asString()` step will no longer allow `null` input. An `IllegalArgumentExce
 
 See: link:https://lists.apache.org/thread/q76pgrvhprosb4lty63bnsnbw2ljyl7m[DISCUSS] thread
 
+==== Removal of has(key, traversal)
+
+The current `has(key, traversal)` implementation has long caused confusion, as it only checks if the traversal produces
+any result, rather than performing equality comparison with the traversal's output value. While fixing this behavior is
```

Comment:
> While fixing this behavior is desirable, it would require extensive changes.

This might be a bit of a personal nit but I'd prefer to keep sentiments such as "we'd like to fix this but not for now" out of the upgrade docs. While I do think that's accurate as we would like reintroduce this once we can do something like `P.eq(Traversal))`, I don't think that context is helpful here.

I think that for users, it's enough that we advertise this change based on its own merits of cleaning up the confusion due to all other forms of `has()` being equality based, as well as eliminating the overlap with `where(Traversal)`.